### PR TITLE
[Tests] - Add tests for GenerationPlugin

### LINKED ISSUE #50: Flavor Dimensions not supported
https://github.com/vanniktech/gradle-android-junit-jacoco-plugin/blob/master/src/main/groovy/com/vanniktech/android/junit/jacoco/Generation.groovy#L93

It looks like you may need to iterate over the "variants" in order to pick up the flavor dimensions.

### PR DESCRIPTION
**Pull request by jaredsburrows**
@vanniktech @passsy Helps close #50 
- Renamed Generation to GenerationPlugin (since it is a plugin)
- Add tests for all tasks that a generation
- Instead of downloading the SDK on each build, add the files that are "needed" by the plugin

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# EditorConfig helps developers define and maintain consistent
+# coding styles between different editors and IDEs
+# editorconfig.org
+
+# top-most EditorConfig file
+root = true
```
Comment (jaredsburrows) [inline .editorconfig]: Added this from Apk Size Plugin.

File: build.gradle [hunk 2]
```diff
@@ -27,6 +27,7 @@ dependencies {
     compile 'com.android.tools.build:gradle:2.3.3'
 
     testCompile 'junit:junit:4.12'
+    testCompile 'org.spockframework:spock-core:1.1-groovy-2.4', { exclude module: "groovy-all" } // Use localGroovy()
```
Comment (jaredsburrows) [inline build.gradle]: Love Spock.

**Comment by jaredsburrows** [comment]
@vanniktech LGTM?

**Comment by vanniktech** [comment]
Cool, thanks, I'll have a look tomorrow.

**Comment by jaredsburrows** [comment]
@vanniktech Just click "merge" :)

File: src/test/resources/android-sdk/build-tools/25.0.2/source.properties [hunk 3]
```diff
@@ -0,0 +1,7 @@
+### Android Tool: Source of this archive.
+#Sat Dec 17 16:06:35 PST 2016
+Archive.HostOs=macosx
```
Comment (vanniktech) [inline src/test/resources/android-sdk/build-tools/25.0.2/source.properties]: Is this file really needed when the license is specified?

File: src/test/resources/android-sdk/platform-tools/source.properties [hunk 4]
```diff
@@ -0,0 +1,7 @@
+### Android Tool: Source of this archive.
+#Sat Dec 17 16:06:16 PST 2016
+Archive.HostOs=macosx
```
Comment (vanniktech) [inline src/test/resources/android-sdk/platform-tools/source.properties]: Is this file really needed when the license is specified?

**Comment by jaredsburrows** [comment]
@vanniktech I just copied these from the my "android-sdk" folder. This is a minimal amount that I have found to work. I use this in my "license-plugin" here: https://github.com/jaredsburrows/gradle-license-plugin.

**Comment by vanniktech** [comment]
Will try to release a new version today if not on this weekend.

**Comment by jaredsburrows** [comment]
@vanniktech Thanks. Currently I am not using "flavor dimensions" currently but I like supporting good projects like this!
